File: src/main/java/com/github/kindrat/cassandra/client/configuration/ApplicationConfiguration.java
package com.github.kindrat.cassandra.client.configuration;

import com.github.kindrat.cassandra.client.service.BackgroundTaskExecutor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.support.ReloadableResourceBundleMessageSource;

@Configuration
public class ApplicationConfiguration {

    @Bean
    public ReloadableResourceBundleMessageSource messageSource() {
        ReloadableResourceBundleMessageSource messageSource = new ReloadableResourceBundleMessageSource();
        messageSource.setBasename("classpath:i18n/messages");
        return messageSource;
    }

    @Bean
    public BackgroundTaskExecutor backgroundTaskExecutor() {
        return new BackgroundTaskExecutor();
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/configuration/ApplicationConfiguration.kt
package com.github.kindrat.cassandra.client.configuration

import com.github.kindrat.cassandra.client.service.BackgroundTaskExecutor
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.support.ReloadableResourceBundleMessageSource

@Configuration
class ApplicationConfiguration {
    @Bean
    fun messageSource(): ReloadableResourceBundleMessageSource {
        val messageSource = ReloadableResourceBundleMessageSource()
        messageSource.setBasename("classpath:i18n/messages")
        return messageSource
    }

    @Bean
    fun backgroundTaskExecutor(): BackgroundTaskExecutor {
        return BackgroundTaskExecutor()
    }
}